### ISSUE TITLE
[Custom Domain] Verification of custom from address across regions 

### ISSUE THREAD

**Issue by jeantanzj** [opengovsg/postmangovsg#759]
In #728 , Keewei pointed out that the workers and backend might use ses credentials in different regions. So we need to check that the custom from address is verified on both.

This problem can be resolved once we release #711 as workers and backend will share secrets.

**Comment by lamkeewei** [comment]
If I remembered correctly, we previously set different SES credentials for OTPs (backend) and workers to ensure that users can still login even in the case our sending SES creds get blocked by high bounces. So just a note that this redundancy will be removed once we share SES creds on workers and backend.

**Comment by jeantanzj** [comment]
Not going to remove the redudancy.  It's just that since they share secrets, backend can access worker's secrets and validate the custom from address using  workers' secrets

**Comment by lamkeewei** [comment]
Re-opening this. After #910 is merged, we need to implement the DKIM checks for both AWS regions used in backend and worker's SES credentials. 

Currently, the checks are only for the regions used in backend's SES credentials.

**Comment by tan-zx** [comment]
Picking up this issue:
- Right now the worker and backend call two different SES regions to send out emails but the env variables are set up such that each has access to the env variables for both regions
- Currently, only the backend verifies the custom domain with its own AWS SES region. This is not great because (1) the actual sending is done by the worker, which uses a different region; (2) while, currently, as a matter of practice, we add every custom domain to both regions, there is a small chance where we might screw up and add it to one and not the other.

As such, the solution is for the backend to validate across both regions.
- Currently, when workers send out emails, they do not verify the custom address with AWS SES. this is fine as long as the backend as already done the validation at an earlier step, which will surface any such error earlier
- We might switch between the two regions if our bounce rates get too high

To-Dos:
- [ ] Change config schema to make clear env variables for worker vs backend
- [ ] Use backend to validate email with both set of AWS regions
- [ ] Check to see if a similar verification process exists for API emails
